compiler/rustc_middle/src/mir/interpret/allocation.rs

```diff
@@ -243,16 +241,16 @@ impl<Prov> Allocation<Prov> {
     ///
     /// If `panic_on_fail` is true, this will never return `Err`.
     pub fn uninit<'tcx>(size: Size, align: Align, panic_on_fail: bool) -> InterpResult<'tcx, Self> {
-        let count128 = (size.bytes_usize() / 16) + 1;
+        let align_usize: usize = align.bytes().try_into().unwrap();
+        let count_align = ((size.bytes_usize() / align_usize) + 1)*align_usize;
+
         // TODO this one is supposed to handle allocation failure, so do so.
```

Comment:
This TODO should be completed before we send it upstream - look up `MaybeUninit` and ask me if you get stuck.